File: frontend/app/src/types/analytics.ts
import type { SatelliteDetection } from "./dashboard";

export interface ZoneStats {
  name: string;
  total_detections: number;
  matched_count: number;
  unmatched_count: number;
  match_rate: number;
  dark_vessels: number;
  alert_count: number;
  critical_alerts: number;
  avg_risk_score: number;
}

export interface NameCount {
  name: string;
  count: number;
}

export interface ZoneAlert {
  id: number;
  mmsi: number;
  vessel_name: string;
  severity: "CRITICAL" | "WARNING" | "INFO";
  reason: string;
  confidence: number;
  lat: number;
  lon: number;
  status: string;
  created_at: string;
}

export interface ZoneDetail extends ZoneStats {
  vessel_type_breakdown: NameCount[];
  alert_severity_breakdown: NameCount[];
  recent_alerts: ZoneAlert[];
  recent_detections: SatelliteDetection[];
}

export interface AnalyticsOverview {
  total_detections: number;
  overall_match_rate: number;
  highest_risk_zone: string;
  zones_covered: number;
  total_dark_vessels: number;
  total_alerts: number;
}

export type ZoneMetric = "detections" | "dark_vessels" | "alerts" | "match_rate";
export type TimeRange = "24" | "48" | "168";

File: backend/internal/model/analytics.go
package model

type ZoneStats struct {
	Name            string  `json:"name"`
	TotalDetections int     `json:"total_detections"`
	MatchedCount    int     `json:"matched_count"`
	UnmatchedCount  int     `json:"unmatched_count"`
	MatchRate       float64 `json:"match_rate"`
	DarkVessels     int     `json:"dark_vessels"`
	AlertCount      int     `json:"alert_count"`
	CriticalAlerts  int     `json:"critical_alerts"`
	AvgRiskScore    float64 `json:"avg_risk_score"`
}

type ZoneDetail struct {
	ZoneStats
	VesselTypeBreakdown    []NameCount          `json:"vessel_type_breakdown"`
	AlertSeverityBreakdown []NameCount          `json:"alert_severity_breakdown"`
	RecentAlerts           []Alert              `json:"recent_alerts"`
	RecentDetections       []SatelliteDetection `json:"recent_detections"`
}

type NameCount struct {
	Name  string `json:"name"`
	Count int    `json:"count"`
}

type AnalyticsOverview struct {
	TotalDetections  int     `json:"total_detections"`
	OverallMatchRate float64 `json:"overall_match_rate"`
	HighestRiskZone  string  `json:"highest_risk_zone"`
	ZonesCovered     int     `json:"zones_covered"`
	TotalDarkVessels int     `json:"total_dark_vessels"`
	TotalAlerts      int     `json:"total_alerts"`
}
